### PR TITLE
Reorder text processing, tweak mention format, escape tilde

### PR DESCRIPTION
**Pull request by mocobeta**
#89 

I think this should be safe.

1. Convert Jira's `[~username]` mention to GitHub mention.
2. Escape all `~` in Jira data
3. Convert Jira syntax to Markdown

### CODE REVIEW

**Comment by mocobeta** [comment]
The result seems fine: 
https://github.com/mocobeta/migration-test-3/issues/522

Screenshot from 2022-08-04 00-17-21

File: migration/src/jira_util.py [hunk 1]
```diff
@@ -287,9 +276,23 @@ def escape_gh_issue_link(m: re.Match):
             disp_name = jira_users.get(jira_id)
             gh_m = account_map.get(jira_id)
             # replace Jira name with GitHub account or Jira display name if it is available, othewise show Jira name with ``
-            mention = lambda: f"@{gh_m}" if gh_m else disp_name if disp_name else f"`~{jira_id}`"
+            mention = lambda: f"@{gh_m}" if gh_m else disp_name if disp_name else f"`@{jira_id}`"
```
Comment (mocobeta) [inline migration/src/jira_util.py:279]: Use `@` instead of `~` for mentions.

File: migration/src/jira_util.py [hunk 2]
```diff
@@ -287,9 +276,23 @@ def escape_gh_issue_link(m: re.Match):
...
             text = text.replace(m, mention())
     
+    # escape tilde to avoid unintentional strike-throughs in Markdown
+    text = text.replace("~", "\~")
```
Comment (mocobeta) [inline migration/src/jira_util.py:283]: Escape `~` before Markdown conversion and after mention conversion.

File: migration/src/jira_util.py [hunk 3]
```diff
@@ -287,9 +276,23 @@ def escape_gh_issue_link(m: re.Match):
...
+    # convert Jira markup into Markdown with customization
+    elements = MarkupElements()
+    elements.replace(UnorderedList, UnorderedTweakedList)
+    elements.replace(OrderedList, OrderedTweakedList)
+    elements.replace(BlockQuote, TweakedBlockQuote)
+    elements.replace(Quote, TweakedQuote)
+    elements.replace(Monospaced, TweakedMonospaced)
+    elements.insert_after(Ruler, LongRuler)
+    elements.append(EscapeHtmlTag)
+    text = jira2markdown.convert(text, elements=elements)
```
Comment (mocobeta) [inline migration/src/jira_util.py:294]: Do syntax conversion after escaping `~`.

**Review by mikemccand — APPROVED** [review]

**Pull request merged**